src/lang.ts

```diff
@@ -10,95 +10,55 @@ function isObject(item: any): boolean {
 	return item && typeof item === 'object' && !Array.isArray(item) && !(item instanceof RegExp);
 }
 
-function copyArray(array: any[], kwArgs: CopyArgs): any[] {
+function copyArray(array: any[], inherited: boolean): any[] {
 	return array.map(function (item: any): any {
 		if (Array.isArray(item)) {
-			return copyArray(item, kwArgs);
+			return copyArray(item, inherited);
 		}
 
-		return isObject(item) ?
-			copy({
+		return !isObject(item) ?
+			item :
+			_mixin({
+				deep: true,
+				inherited: inherited,
 				sources: [ item ],
-				deep: kwArgs.deep,
-				descriptors: kwArgs.descriptors,
-				inherited: kwArgs.inherited,
-				assignPrototype: kwArgs.assignPrototype
-			}) :
-			item;
+				target: Object.create(null)
+			});
 	});
 }
 
-export function copy(kwArgs: CopyArgs): any {
-	const sources: Hash<any>[] = kwArgs.sources;
-	let target: any;
-
-	if (!sources.length) {
-		throw new RangeError('lang.copy requires at least one source object.');
-	}
-
-	if (kwArgs.assignPrototype) {
-		// create from the same prototype
-		target = Object.create(Object.getPrototypeOf(sources[0]));
-	}
-	else {
-		// use the target or create a new object
-		target = kwArgs.target || {};
-	}
+interface MixinArgs {
+	deep: boolean;
+	inherited: boolean;
+	sources: {}[];
```

Comment:
Interestingly, native `Object.assign` doesn't discriminate. For example, run `Object.assign(new Date(), /\s/)` in Firefox. I haven't tested whether it does anything useful, but at the very least it doesn't throw an error. As for the typing, I personally think it's fine. It's explicit that objects are meant to be used, and it doesn't require the user to throw in additional coercions just to use it.